mongo/integration/sessions_test.go

```diff
@@ -475,6 +476,40 @@ func TestSessions(t *testing.T) {
 		assert.True(mt, limitedSessionUse, limitedSessMsg, len(ops))
 
 	})
+
+	// Regression test for GODRIVER-2533. Note that this test assumes the race
+	// detector is enabled (GODRIVER-2072).
+	mt.Run("NumberSessionsInProgress data race", func(mt *mtest.T) {
+		// Use a channel to sync a few simultaneous runs of NumberSessionsInProgress
+		// and a basic collection operation (CountDocuments).
+		ch := make(chan struct{})
+
+		// Use a WaitGroup to ensure that both goroutines are done executing before
+		// running test cleanup.
+		var wg sync.WaitGroup
+		wg.Add(2)
+
+		go func() {
+			defer wg.Done()
+
+			<-ch
+			for i := 0; i < 10; i++ {
+				_ = mt.Client.NumberSessionsInProgress()
+			}
+		}()
+		go func() {
+			defer wg.Done()
+
+			<-ch
+			for i := 0; i < 10; i++ {
+				_, err := mt.Coll.CountDocuments(context.Background(), bson.D{})
+				assert.Nil(mt, err, "CountDocument error: %v", err)
+			}
+		}()
+
+		close(ch)
```

Comment:
Optional: This channel close synchronization likely isn't doing much because of the typical delay between creating goroutines and when they start running on a thread. Consider removing it.

To elaborate, the most likely sequence of events is something like:
1. Create `ch` channel.
2. Create `NumberSessionsInProgress` goroutine.
3. Create `CountDocuments` goroutine.
4. Close `ch` channel.
5. `NumberSessionsInProgress` goroutine starts running.
6. `CountDocuments` goroutine starts running.